supervisor/api/addons.py

```diff
@@ -412,6 +412,54 @@ def logs(self, request: web.Request) -> Awaitable[bytes]:
         addon = self._extract_addon(request)
         return addon.logs()
 
+    @api_process_raw(CONTENT_TYPE_PNG)
+    async def icon(self, request: web.Request) -> bytes:
+        """Return icon from add-on."""
+        addon = self._extract_addon(request)
+        if not addon.with_icon:
+            raise APIError(f"No icon found for add-on {addon.slug}!")
+
+        with addon.path_icon.open("rb") as png:
+            return png.read()
+
+    @api_process_raw(CONTENT_TYPE_PNG)
+    async def dark_icon(self, request: web.Request) -> bytes:
+        """Return dark icon from add-on."""
+        addon = self._extract_addon(request)
+        if not addon.with_icon:
+            raise APIError(f"No icon found for add-on {addon.slug}!")
+
+        if addon.with_dark_icon:
+            with addon.path_dark_icon.open("rb") as png:
+                return png.read()
+
+        with addon.path_icon.open("rb") as png:
+            return png.read()
+
+    @api_process_raw(CONTENT_TYPE_PNG)
+    async def logo(self, request: web.Request) -> bytes:
+        """Return logo from add-on."""
+        addon = self._extract_addon(request)
+        if not addon.with_logo:
+            raise APIError(f"No logo found for add-on {addon.slug}!")
+
+        with addon.path_logo.open("rb") as png:
+            return png.read()
+
+    @api_process_raw(CONTENT_TYPE_PNG)
+    async def dark_logo(self, request: web.Request) -> bytes:
+        """Return dark logo from add-on."""
+        addon = self._extract_addon(request)
+        if not addon.with_logo:
+            raise APIError(f"No logo found for add-on {addon.slug}!")
+
+        if addon.with_dark_logo:
+            with addon.path_dark_logo.open("rb") as png:
+                return png.read()
+
+        with addon.path_logo.open("rb") as png:
+            return png.read()
+
```

Comment:
We should readd this if we change the logic and cache it after installation to keep it same also if the store change it until next update. For now, we can just route over the legacy part until we add such logick

Reply:
See response above.
Adding new endpoints to "legacy" is not right, especially here where they _should_ have a different meaning.